addons/mail/views/mail_template_views.xml

```diff
@@ -121,7 +121,7 @@
                 <search string="Templates">
                     <field name="name" filter_domain="['|', '|', ('name','ilike',self), ('subject','ilike',self), ('email_to','ilike',self)]" string="Templates"/>
                     <field name="lang"/>
-                    <field name="model_id"/>
+                    <field name="model_id" filter_domain="[('model_id', '=', raw_value)]"/>
```

Comment:
What is that thing ? generally we use "self", and anyway don't see the added value ?

Reply:
For the field domain:
- `self` represents the string typed by the user when using the search bar. Unfortunately, I can not set that value from the action context. It's something that is internal to the search model.
- `raw_value` represents the "search_default_..." value passed to the context. So, in my code, when I set `search_default_model_id: 42`, `raw_value` will be equal to `42` for the filter domain set on the field `model_id`.
The magic values are set here: https://github.com/odoo/odoo/blob/d6c30715d21b86179662578741aa1423a7730f10/addons/web/static/src/search/search_model.js#L1671-L1674

Ideally, it would be better to do something like: `filter_domain="[('model_id.name', '=', raw_value)]" to avoid adding a new field to get the model id but I get an exception when doing that (like: error in clause "... WHERE id in ('k', 'n', 'o', ...)"). That's a bit weird. It's possible that this thing works by accident.

The filter domain allows us to pre-select all templates that are linked to the model you were on. We used that filter in particular for the display: It shows "Applies to: Knowledge Article" which is self explanatory and easy for the user to understand.

Reply:
Update: I will search on the `model` field.